installer/mod.ts

```diff
@@ -1,25 +1,34 @@
 #!/usr/bin/env deno --allow-all
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-const {
-  args,
-  env,
-  readDirSync,
-  mkdirSync,
-  writeFile,
-  exit,
-  stdin,
-  chmod,
-  remove,
-  run
-} = Deno;
+const { env, stdin, args, exit, writeFile, chmod, run } = Deno;
+import { parse } from "../flags/mod.ts";
 import * as path from "../fs/path.ts";
 import { exists } from "../fs/exists.ts";
+import { ensureDir } from "../fs/ensure_dir.ts";
 
 const encoder = new TextEncoder();
 const decoder = new TextDecoder("utf-8");
-const isWindows = Deno.platform.os === "win";
 // Regular expression to test disk driver letter. eg "C:\\User\username\path\to"
 const driverLetterReg = /^[c-z]:/i;
+const isWindows = Deno.platform.os === "win";
+
+function showHelp(): void {
+  console.log(`deno installer
+  Install remote or local script as executables.
+
+USAGE:
+  deno -A https://deno.land/std/installer/mod.ts [OPTIONS] EXE_NAME SCRIPT_URL [FLAGS...]
+
+ARGS:
+  EXE_NAME  Name for executable
+  SCRIPT_URL  Local or remote URL of script to install
+  [FLAGS...]  List of flags for script, both Deno permission and script specific
+              flag can be used.
+              
+OPTIONS:
+  -d, --dir <PATH> Installation directory path (defaults to ~/.deno/bin)
+`);
```

Comment:
I wish the flags module was more like Go's... This would be generated automatically.

Reply:
Agreed, I was looking at porting one of more advanced CLI modules from node:
- https://github.com/yargs/yargs
- https://github.com/tj/commander.js/blob/master/index.js
- https://github.com/mattallty/Caporal.js

But I might as well look into https://golang.org/pkg/flag/